tests/pipelines/stable_diffusion/test_stable_diffusion_controlnet.py

```diff
@@ -585,6 +586,42 @@ def test_canny_guess_mode(self):
         expected_slice = np.array([0.2724, 0.2846, 0.2724, 0.3843, 0.3682, 0.2736, 0.4675, 0.3862, 0.2887])
         assert np.abs(image_slice.flatten() - expected_slice).max() < 1e-2
 
+    def test_stable_diffusion_compile(self):
+        if version.parse(torch.__version__) < version.parse("2.0"):
+            print(f"Test `test_stable_diffusion_ddim` is skipped because {torch.__version__} is < 2.0")
+            return
+
+        controlnet = ControlNetModel.from_pretrained("lllyasviel/sd-controlnet-canny")
+
+        pipe = StableDiffusionControlNetPipeline.from_pretrained(
+            "runwayml/stable-diffusion-v1-5", safety_checker=None, controlnet=controlnet
+        )
+        pipe.to("cuda")
```

Comment:
does this pass test? I am getting a 
```
  File "/home/yj/pytorch/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/yj/diffusers/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_controlnet.py", line 892, in __call__
    self.check_inputs(
  File "/home/yj/diffusers/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_controlnet.py", line 984, in <resume in __call__>
    latents = self.prepare_latents(
  File "/home/yj/diffusers/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_controlnet.py", line 734, in prepare_latents
    latents = randn_tensor(shape, generator=generator, device=device, dtype=dtype)
  File "/home/yj/diffusers/src/diffusers/utils/torch_utils.py", line 75, in randn_tensor
    latents = torch.randn(shape, generator=generator, device=rand_device, dtype=dtype, layout=layout).to(device)
RuntimeError: Expected a 'cuda' device type for generator but found 'cpu'
```
after rebasing with this change. Thanks!